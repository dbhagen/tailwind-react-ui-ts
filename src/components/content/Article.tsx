import React from 'react'
import PropTypes from 'prop-types'

import ContentBlock from './ContentBlock'

const Article = ({ is, ...rest }) => <ContentBlock is={is} {...rest} />

Article.propTypes = {
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
}

Article.defaultProps = {
  is: 'article',
}

export default Article
