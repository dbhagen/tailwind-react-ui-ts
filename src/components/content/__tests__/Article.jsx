import React from 'react'
import { shallow } from 'enzyme'

import Article from '../Article'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)

  const wrapper = shallow(<Article {...props}>Lorem ipsum...</Article>)

  return {
    props,
    wrapper,
  }
}

describe('Article', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
