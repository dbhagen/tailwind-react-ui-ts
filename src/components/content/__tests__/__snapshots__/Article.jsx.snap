// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders matching snapshot 1`] = `
<WithTheme(ContentBlock)
  is="article"
>
  Lorem ipsum...
</WithTheme(ContentBlock)>
`;
